packages/react-dom/src/client/ReactDOMComponentTree.js

```diff
@@ -18,9 +18,9 @@ import {getParentSuspenseInstance} from './ReactDOMHostConfig';
 const randomKey = Math.random()
   .toString(36)
   .slice(2);
-const internalInstanceKey = '__reactInternalInstance$' + randomKey;
```

Comment:
https://github.com/facebook/react/blob/0545f366d4d6b5959f4bb172e810c745f74b9513/packages/react-devtools-shared/src/backend/views/Highlighter/Overlay.js#L284

Reply:
We should never do these hacks in DevTools, we should use the injected APIs to do it or add a new injected API if we don't have one.

Reply:
I think this (the overlay plugin) was ported without a ton of refactoring from the v3 codebase.

The nice thing about DevTools is that it lives in the React repo so we can clean any hacks like this up as needed 😄

#17822

Reply:
P.S. #17841